03week/mastermind.js

```diff
@@ -42,26 +42,26 @@ function generateHint(userGuess) {
       userGuessArr[i]= '';
     }
   }
- for(let i=0; i<userGuessArr.length; i++) {
-   if(userGuessArr[i]) {
-     for (let j=0; j<solutionArr.length; j++) {
-       if (userGuessArr[i] === solutionArr[j]) {
-         correctLetter++;
-         solutionArr[j] = '';
-       }
-     }
-   }
- }
+  for(let i=0; i<userGuessArr.length; i++) {
+    if(userGuessArr[i]) {
+      for (let j=0; j<solutionArr.length; j++) {
+        if (userGuessArr[i] === solutionArr[j]) {
+          correctLetter++;
```

Comment:
You can use the `indexOf()` js method here to consolidate your code here and avoid a double `for` loop.